xmodule/capa/capa_problem.py

```diff
@@ -328,6 +336,7 @@ def get_state(self):
                 'student_answers': self.student_answers,
                 'has_saved_answers': self.has_saved_answers,
                 'correct_map': self.correct_map.get_dict(),
+                'correct_map_history': [cmap.get_dict() for cmap in self.correct_map_history],
```

Comment:
I wonder why we have to generate this list every time we get the user session. Why can't we use `self.correct_map_history`? Why don't we save cmap.get_dict directly instead there?

Reply:
For the same reason that must be done for the `correct_map` field. For this case, we need to get each **correct_map** in the list as a dictionary, but at other times we need it as a `CorrectMap` object.